### ISSUE TITLE
Add ability to restore specifically from other wallet, in addition to generic restore

### ISSUE THREAD

**Issue by esengulov** [horizontalsystems/unstoppable-wallet-android#5724]
Certain wallets have specific implementations i.e. the way multi accounts derived pin Metamask.

Target all major wallets and provide restore for them. That would be easier for onboarding users from those platforms.

**Comment by esengulov** [comment]
> Is derivation path specification already supported in restoring mode?

Yes, the restore mode would take into account custom derivation paths used by popular wallets. Unfortunately, most popular wallets have some specifics. For instance, the way Metamask generates multi accounts is their own invention and not really standard. 

When you take Solana for instance both Trust and Phantom are popular wallets and yet differ in derivation paths for the Solana chain. Trust is particular uses different paths for 12-24 words and for 15 words :) That might be a bug on their end which they don't know about or simply ignoring it.

**Comment by OneNewSmartIdea** [comment]
> 

MetaMask uses the same method ( BIP32 Extended Key) as Ian Coleman open source project.
https://github.com/iancoleman/bip39
https://iancoleman.io/bip39/
